### PR TITLE
Bugfix FXIOS-7697 ⁃ Downloading PDF of Accounts Recovery Key breaks Accounts Recovery flow

### PR DESCRIPTION
**Pull request by dicarobinho**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-7697)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/17175)

## :bulb: Description
Presenting a controller in order to save the PDF file from Mozilla Account (Recovery Key)
We are presenting that controller only if the page is a blob, is from the mozilla accounts and it can be converted to PDF.

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

File: firefox-ios/RustFxA/FxAWebViewModel.swift [hunk 1]
```diff
@@ -149,6 +172,53 @@ class FxAWebViewModel: FeatureFlaggable {
...
+            }
+
+            pdf.write(to: outputURL)
+            if FileManager.default.fileExists(atPath: outputURL.path) {
+                let url = URL(fileURLWithPath: outputURL.path)
+                return url
+            }
+
+            return nil
+        case .failure: return nil
```
Comment (nbhasin2) [inline firefox-ios/RustFxA/FxAWebViewModel.swift:219]: nit: add log for failure

File: firefox-ios/Shared/Extensions/WKWebViewExtensions.swift [hunk 2]
```diff
@@ -43,6 +43,26 @@ extension WKWebView {
...
+                                                         completion: @escaping (Result<Any?, Error>) -> Void ) {
+        self.callAsyncJavaScript(script,
+                                 arguments: arguments,
+                                 in: nil,
+                                 in: .defaultClient) { result in
+            switch result {
+            case .success(let value):
+                completion(.success(value))
+            case .failure(let error):
+                completion(.failure(error))
```
Comment (nbhasin2) [inline firefox-ios/Shared/Extensions/WKWebViewExtensions.swift:62]: nit: log for failure to capture the error

File: firefox-ios/firefox-ios-tests/Tests/ClientTests/FxAWebViewModelTests.swift [hunk 3]
```diff
@@ -0,0 +1,122 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+@testable import Client
+import XCTest
+import PDFKit
+
+// swiftlint:disable force_try
```
Comment (nbhasin2) [inline firefox-ios/firefox-ios-tests/Tests/ClientTests/FxAWebViewModelTests.swift:9]: @dicarobinho, do you really need this as my guess is its only for 

https://github.com/mozilla-mobile/firefox-ios/pull/20215/files#diff-bf65ed39518f585b6ff439e5e20ad51766a498496762890aebc5ade1950534eeR114-R115

Comment (nbhasin2) [reply firefox-ios/firefox-ios-tests/Tests/ClientTests/FxAWebViewModelTests.swift:9]: And other cases, if we can improve the force try then that would be amazing

File: firefox-ios/RustFxA/FxAWebViewModel.swift [hunk 4]
```diff
@@ -42,6 +43,28 @@ class FxAWebViewModel: FeatureFlaggable {
...
+                      resolve(reader.result);
+                    };
+                    reader.onerror = reject;
+                    reader.readAsDataURL(blob);
+                  });
+                }
+
+                const url = await createBlobFromUrl(blobUrl)
+                return await blobToDataURLAsync(url)
+            """
```
Comment (nbhasin2) [inline firefox-ios/RustFxA/FxAWebViewModel.swift:66]:
```suggestion
            async function createBlobFromUrl(url) {
              const response = await fetch(url);
              const blob = await response.blob();
              return blob;
            }

            function blobToDataURLAsync(blob) {
              return new Promise((resolve, reject) => {
                const reader = new FileReader();
                reader.onload = () => {
                  resolve(reader.result);
                };
                reader.onerror = reject;
                reader.readAsDataURL(blob);
              });
            }

            const url = await createBlobFromUrl(blobUrl)
            return await blobToDataURLAsync(url)
        """
```
nit: indentation

**Review by nbhasin2 — APPROVED** [review]

File: firefox-ios/RustFxA/FxAWebViewModel.swift (hunk 1 above)
Comment (dicarobinho) [reply firefox-ios/RustFxA/FxAWebViewModel.swift]: Added, resolved.

File: firefox-ios/Shared/Extensions/WKWebViewExtensions.swift (hunk 2 above)
Comment (dicarobinho) [reply firefox-ios/Shared/Extensions/WKWebViewExtensions.swift:62]: No logs required here.

File: firefox-ios/firefox-ios-tests/Tests/ClientTests/FxAWebViewModelTests.swift (hunk 3 above)
Comment (dicarobinho) [reply firefox-ios/firefox-ios-tests/Tests/ClientTests/FxAWebViewModelTests.swift]: Fixed them.

**Comment by isabelrios** [comment]
Please rebase to get the swiftlint fix from main. Thanks!

**Pull request merged**